src/audio/dai-zephyr.c

```diff
@@ -873,21 +874,24 @@ static int dai_set_dma_buffer(struct dai_data *dd, struct comp_dev *dev,
 		comp_err(dev, "dai_set_dma_buffer(): no valid dma buffer period count");
 		return -EINVAL;
 	}
-
 	buffer_size = ALIGN_UP(period_count * period_bytes, align);
+	period_count = MAX(period_count,
+			   SOF_DIV_ROUND_UP(dd->ipc_config.dma_buffer_size, period_bytes));
 	*pc = period_count;
```

Comment:
should this line be deleted now? Also in general better not to modify it if we hit an error below